### DISCUSSION TITLE
Automatic fishing

### DISCUSSION THREAD

**Discussion by Starjumper2821** [TechReborn/TechReborn#2526] [Ideas]
**Is your feature request related to a problem? Please describe.**
Fishing has long been the best way to gain enchanted books as well as black dye. To fish requires player interaction or at best afk farms, though the effectiveness of those has been reduced in 1.16.

**Describe the solution you'd like**
A machine much like the greenhouse controller that administers a section of water following the open water rules introduced in 20w13a. Overclocker upgrades might act like a lure enchantment, speeding up the intervals in which an item is ejected.

I'd consider the recipe of the greenhouse controller with the hoes replaced by fishing rods to be a good balance.

**Describe alternatives you've considered**
Minefactory Reloaded had a Fisher but it only produced cod due to being coded before other fish variants or treasure were introduced.

**Comment by saltyseadoggo** [comment]
Mojang nerfed AFK fish farms in 1.16 because they were quite an overpowered way of obtaining enchanted books with essentially no effort beyond the minimal initial farm setup. They make the likes of villager trading halls entirely obsolete, because why would you bother faffing about with villagers when you can just AFK for a while to gather more books than you'll likely ever need? A TR machine to replace them would be just as broken, if not more due to running whilst the player can do other things; it would be idle enchanted book generation that only costs FE power. Fishing as a primary source of enchanted books doesn't make much sense anyways, compared to buying them from librarian villagers that actually deal in books.
